### PR TITLE
Proposed fix for refreshNodeInfo

### PR DESCRIPTION
**Pull request by carsten19**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -919,7 +919,7 @@ function extendChannel(nodeID, comClass, valueId) {
             if (role && !stateObj.common.role) stateObj.common.role = role;
 
             stateObj.native = valueId;
-            adapter.log.info('State updated: ' + stateID + ' = ' + value + ', index = ' + valueId.index + ', comClass = ' + comClass + ', instance = ' + valueId.instance);
+            adapter.log.debug('State updated: ' + stateID + ' = ' + value + ', index = ' + valueId.index + ', comClass = ' + comClass + ', instance = ' + valueId.instance);
```
Comment (AlCalzone) [inline main.js:922]: What's the logic behind these loglevel changes? Some went from info to debug, some the other way round?

File: main.js [hunk 2]
```diff
@@ -1284,23 +1284,22 @@ function main() {
     });
 
     zwave.on('value removed', function (nodeID, comClass, instance, index) {
-        adapter.log.info('value removed: ' + nodeID + ' comClass: ' + JSON.stringify(comClass) + ' instance ' + instance + ' value: '  + JSON.stringify(index));
+        adapter.log.debug('value removed: ' + nodeID + ' comClass: ' + JSON.stringify(comClass) + ' instance ' + instance + ' value: '  + JSON.stringify(index));
 
-        deleteDevice(nodeID, function() { //why here and not on "node removed" ?
```
Comment (AlCalzone) [inline main.js:1289]: Ahh, I see. I had this question before. Good to know that deleting the device here is indeed not the desired behavior.

File: main.js [hunk 3]
```diff
@@ -1284,23 +1284,22 @@ function main() {
...
-            //disable exclusion automatically after 5 seconds
-            setTimeout(function(){disableExclusion();}, 5000);
-        });
+        var id = calcName(nodeID, comClass, index, instance);
```
Comment (AlCalzone) [inline main.js]: We should use `let` and `const` moving forward. So this should be `const id = ...`.

File: main.js [hunk 4]
```diff
@@ -1284,23 +1284,22 @@ function main() {
...
+        var list = [ id ];
+        delObjects(list);
```
Comment (AlCalzone) [inline main.js]: Please remove the list variable and change this to `delObjects([id])`.

File: package.json [hunk 5]
```diff
@@ -40,7 +40,7 @@
     "url": "git+https://github.com/ioBroker/ioBroker.zwave.git"
   },
   "dependencies": {
-    "openzwave-shared": "^1.5.5",
+    "openzwave-shared": "^1.5.6",
```
Comment (AlCalzone) [inline package.json:43]: This is unnecessary, since `^1.5.5` also includes version 1.5.6. But it doesn't hurt either.

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Just two small changes and some clarification please ;)

File: main.js (hunk 1 above)
Comment (carsten19) [reply main.js:922]: I had a look at the log statements to verify what is actually necessary. For example state changes are normal behavior and clutter the log file. But if functions are called which are not implemented (like polling) that should be visible.

File: main.js (hunk 2 above)
Comment (carsten19) [reply main.js:1289]: Well the node is refreshed and reported as new. But the value changes are reported individually by the openzwave-shared module. That's why you see tons of delete entries in the bug report.
The change from 1.4 to 1.6 is that it actually **does** refresh as the openzwave cache is dropped.

File: main.js (hunk 4 above)
Comment (carsten19) [reply main.js]: Oki doki :)

File: main.js (hunk 2 above)
Comment (carsten19) [reply main.js:1289]: To add to that: what I could not test so far if it also works when that value removed is fired without refreshing so when the values change "normally". I simply don't have an idea about a use case for it.

File: main.js [hunk 6]
```diff
@@ -1286,9 +1286,8 @@ function main() {
     zwave.on('value removed', function (nodeID, comClass, instance, index) {
         adapter.log.debug('value removed: ' + nodeID + ' comClass: ' + JSON.stringify(comClass) + ' instance ' + instance + ' value: '  + JSON.stringify(index));
 
-        var id = calcName(nodeID, comClass, index, instance);
-        var list = [ id ];
-        delObjects(list);
+        const id = calcName(nodeID, comClass, index, instance);
+        delObjects([list]);
```
Comment (AlCalzone) [inline main.js]: Woops, this should be `id`,  not `list`

**Review by AlCalzone — CHANGES_REQUESTED** [review]
Almost ;)

File: main.js (hunk 6 above)
Comment (carsten19) [reply main.js]: Let's blame it on the heat...

**Review by AlCalzone — APPROVED** [review]

**Pull request merged**